### PR TITLE
Created API endpoints and added bomb SVG for game over screen

### PR DESCRIPTION
**Pull request by koneill6**

### CODE REVIEW

File: Services/SaveGameDAO.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿using Milestone_cst_350.Models;
+using System.Data.SqlClient;
+
+namespace Milestone_cst_350.Services
+{
+    public class SaveGameDAO
+    {
+
+        string connectionString = @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=Test;Integrated Security=True;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False";
```
Comment (cesoun) [inline Services/SaveGameDAO.cs]: We should inherit the `DataService` for access to the `ConnectionString` value.

Any `connectionString` usage should also be updated to `ConnectionString` as well.

File: Services/SaveGameDAO.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        public List<SaveGameModel> AllProducts()
```
Comment (cesoun) [inline Services/SaveGameDAO.cs]: `AllProducts` and related `_products` variables should be changed to be inline with application model naming.

**Review by cesoun — CHANGES_REQUESTED** [review]
Usages of `product` over more application related names such as `game`, `gameModel`, `model`, `...`.

Potentially concerned about `null` return values on functions `AllProducts()`, `GetGameById(...)`, `DeleteProductById(...)`, however not a pressing issue for the most part, ideally just updated function names to be inline w. application. eg. `DeleteGameById(...)`, `AllGames()`.

If possible, the addition of `AllGamesByUserId(...)` would be very useful.

File: Services/SaveGameDAO.cs (hunk 1 above)
Comment (joelsmith-2019) [reply Services/SaveGameDAO.cs]: Fixed

File: Services/SaveGameDAO.cs (hunk 2 above)
Comment (joelsmith-2019) [reply Services/SaveGameDAO.cs]: Fixed

**Comment by joelsmith-2019** [comment]
Just fixed up the previous problems with the branch. I also merged in the latest serialization changes from master. I then added functionality to save game states and retrieve all saved games by a user's id.

**Review by joelsmith-2019 — COMMENTED** [review]
All requested changes have been implemented.

**Review by cesoun — APPROVED** [review]

**Pull request merged**